scripts/setup_sgtm_tasks_project.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+
+import argparse
+from dataclasses import dataclass
+
+import asana
+import sys
+
+
+@dataclass
+class EnumOption:
+    name: str
+    color: str
+
+
+@dataclass
+class CustomField:
+    name: str
+    enum_options: [EnumOption]
+
+
+CUSTOM_FIELDS = [
+    CustomField(
+        name="PR Status",
+        enum_options=[
+            EnumOption(name="Open", color="green"),
+            EnumOption(name="Merged", color="purple"),
+            EnumOption(name="Closed", color="red"),
+        ],
+    ),
+    CustomField(
+        name="Build",
+        enum_options=[
+            EnumOption(name="Success", color="green"),
+            EnumOption(name="Failure", color="red"),
+        ],
+    ),
+]
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="""A helper script to set up an SGTM tasks Asana project with the suggested custom fields
+		"""
+    )
+
+    parser.add_argument(
+        "-p",
+        "--personal_access_token",
+        help="""The Asana Personal Access Token that will be used to set up the Asana project. If you do not already have one,
+		follow the instructions at https://developers.asana.com/docs/personal-access-token
+		""",
+        type=str,
+        required=True,
+    )
+
+    parser.add_argument(
+        "-n",
+        "--new_project_name",
+        help="""The Asana Project name - use this if you want to set up a project from scratch
+		""",
+        type=str,
+        required=False,
+    )
+
+    parser.add_argument(
+        "-t",
+        "--team_id",
+        help="""Only required if you are creating a new project - this is the global ID of the team that
+		you want to share the project with
+		""",
+        type=str,
+        required=False,
+    )
+
+    parser.add_argument(
+        "-e",
+        "--existing_project_id",
+        help="""The global ID of an existing Asana Project - use this if you have an existing Asana project that you want to use 
+		to sync SGTM tasks
+		""",
+        type=str,
+        required=False,
+    )
+
+    return parser.parse_args()
+
+
+def setup_new_project(project_name: str, team_id: str, pat: str) -> None:
+    client = AsanaClient(pat)
+    project_id = client.create_project(project_name, team_id)
+    client.setup_custom_fields(project_id)
+
+
+def setup_existing_project(project_id: str, pat: str) -> None:
+    client = AsanaClient(pat)
+    client.setup_custom_fields(project_id)
+    client.add_user_to_project(project_id)
+
+
+class AsanaClient(object):
+    def __init__(self, personal_access_token: str):
+        self.client = asana.Client.access_token(personal_access_token)
+        self.client.headers = {"Asana-Enable": "string_ids"}
+        self.workspace_id = self._get_workspace_id()
+
+    def _get_workspace_id(self) -> str:
```

Comment:
I'm not fully sold on this so I'm curious to get your opinion - the other option would be to explicitly take in a workspace ID (but that might not be trivial for the user?)

Reply:
This is just picking the first in the list -- what about a more interactive version if there are multiple workspaces and having the user choose from the available workspace ids?

Reply:
Good point - `fzf` might be useful here

Reply:
Would be nice, but I'd prefer not introducing more external dependencies to folks' setup if not necessary

Reply:
Is the alternative to manually write a function that provides a CLI menu sort of thing? That's doable but feels like reinventing the wheel

Reply:
Yeah the alternative is to print out a numbered list and ask for input (1-N)